### PR TITLE
feat: add filter on taskId

### PR DESCRIPTION
**Pull request by esoubiran-aneo**

### CODE REVIEW

File: apps/api/src/app/modules/tasks/tasks.service.ts [hunk 1]
```diff
@@ -35,14 +35,19 @@ export class TasksService implements OnModuleInit {
     limit: number,
     sessionId: string,
     orderBy: string | null,
-    order: string | null
+    order: string | null,
+    _id?: string
```
Comment (lemaitre-aneo) [inline apps/api/src/app/modules/tasks/tasks.service.ts:39]: uniformity

Comment (esoubiran-aneo) [reply apps/api/src/app/modules/tasks/tasks.service.ts:39]: done

File: apps/api/src/app/modules/tasks/tasks.service.ts [hunk 2]
```diff
@@ -35,14 +35,19 @@ export class TasksService implements OnModuleInit {
...
   ): Promise<Pagination<Task>> {
     const startIndex = (page - 1) * limit;
 
-    const match = {
+    const match: { [key: string]: any } = {
       SessionId: sessionId,
     };
 
+    if (_id) {
+      match._id = { $regex: _id, $options: 'i' };
```
[the remark this answers is not in the document]
Comment (esoubiran-aneo) [reply apps/api/src/app/modules/tasks/tasks.service.ts:48]: done

File: apps/api/src/app/modules/tasks/tasks.service.ts [hunk 3]
```diff
@@ -64,14 +69,13 @@ export class TasksService implements OnModuleInit {
       })
       .sort({
         [orderBy || 'StartDate']: (Number(order) as SortOrder) || -1,
-        _id: 1,
+        _id: orderBy === '_id' ? (Number(order) as 1 | -1) : 1,
```
[the remark this answers is not in the document]
Comment (esoubiran-aneo) [reply apps/api/src/app/modules/tasks/tasks.service.ts:72]: done

**Review by lemaitre-aneo — APPROVED** [review]

File: apps/api/src/app/modules/tasks/tasks.service.ts [hunk 4]
```diff
@@ -36,18 +36,32 @@ export class TasksService implements OnModuleInit {
...
 
+    if (_id) {
+      match['_id'] = _id;
+    }
+
     if (status !== undefined) {
       match['Status'] = status;
     }
 
+    const tasksSort: { [key: string]: 1 | -1 } = {};
```
[the remark this answers is not in the document]
Comment (esoubiran-aneo) [reply apps/api/src/app/modules/tasks/tasks.service.ts]: done

File: apps/api/src/app/modules/tasks/tasks.service.ts [hunk 5]
```diff
@@ -36,18 +36,32 @@ export class TasksService implements OnModuleInit {
...
-    const match = {
+    const match: { [key: string]: unknown } = {
       SessionId: sessionId,
     };
 
+    if (taskId) {
+      match['_id'] = taskId;
+    }
+
     if (status !== undefined) {
```
Comment (ngruelaneo) [inline apps/api/src/app/modules/tasks/tasks.service.ts:52]: question bete: en javascript cela ne serait pas equivalent à:

if (status)

Comment (esoubiran-aneo) [reply apps/api/src/app/modules/tasks/tasks.service.ts:53]: done

File: apps/api/src/app/modules/tasks/tasks.service.ts [hunk 6]
```diff
@@ -36,18 +36,32 @@ export class TasksService implements OnModuleInit {
     sessionId: string,
     orderBy: string | undefined,
```
[the remark this answers is not in the document]
Comment (esoubiran-aneo) [reply apps/api/src/app/modules/tasks/tasks.service.ts]: done

**Review by ngruelaneo — APPROVED** [review]

**Review by ngruelaneo — APPROVED** [review]

**Pull request merged**